share/spice/time_and_date_holiday/time_and_date_holiday.js

```diff
@@ -0,0 +1,87 @@
+(function (env) {
+	"use strict";
+	env.ddg_spice_time_and_date_holiday = function(api_result) {
+	    if(!api_result || !api_result.h || api_result.h.length === 0) {
+			return Spice.failed('time_and_date_holiday');
+	    }
+	    var query = DDG.get_query(),
+	        url = 'http://www.timeanddate.com',
+			source = url + '/search/results.html?query=' + encodeURIComponent(query);
+
+        if (api_result.h.length == 1){
+            var holiday = api_result.h[0];
+
+            if (holiday.d.length <= 0) {
+                return Spice.failed('time_and_date_holiday');
+            }
+
+            // selecting the first date only as it's the closest
+
+            Spice.add({
+                id: 'time_and_date_holiday',
+                name: 'Holidays',
+                data: holiday,
+                normalize: function(item) {
+                    var date = item.d[0];
+                    var next_date = '';
+
+                    if (item.d.length > 1) {
+                        next_date = item.d[1];
+                    }
+
+                    return {
+                        title: date,
+                        subtitle: item.n,
+                        description: item.a,
+                        next_date: next_date
+                    };
+                },
+                meta: {
+                    sourceName: 'timeanddate.com',
+                    sourceUrl: url + holiday.u
+                },
+                templates: {
+                    group: 'text',
+                    options: {
+                        content: Spice.time_and_date_holiday.singleday,
+                        moreAt: true
+                    }
+                }
+            });
+        } else {
+            DDG.require('moment.js', function() {
```

Comment:
Got it. I'll fix that. Just curious, is there a way to require moment.js as not a block but globally for a frontend? Like a flag at the top that says moment.js is required. And then we can use moment() anywhere in the main function. Creating blocks seem messy at times. The Flights spice was using variables set in `this` object and using that inside `DDG.require()` wont just work. `this` would point to something else inside that require block. So was just curious.